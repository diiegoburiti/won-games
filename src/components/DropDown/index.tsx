import { useState } from 'react'
import * as S from './styles'

export type DropDownProps = {
  title: React.ReactNode
  children: React.ReactNode
}
const DropDown = ({ title, children }: DropDownProps) => {
  const [isOpen, setIsOpen] = useState(false)

  const handleOpenDropDown = () => {
    setIsOpen(!isOpen)
  }
  return (
    <S.Wrapper isOpen={isOpen}>
      <S.Title onClick={handleOpenDropDown}>{title}</S.Title>

      <S.Content aria-hidden={!isOpen}>{children}</S.Content>
      <S.Overlay aria-hidden={!isOpen} onClick={handleOpenDropDown} />
    </S.Wrapper>
  )
}

export default DropDown
