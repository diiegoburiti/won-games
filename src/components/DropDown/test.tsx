import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithTheme } from 'utils/tests/helpers'

import DropDown from '.'

describe('<DropDown />', () => {
  beforeEach(() => {
    const title = <h5 aria-label="toogle dropdown">Click here</h5>

    const { container } = renderWithTheme(
      <DropDown title={title}>
        <span>content</span>
      </DropDown>
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render title', () => {
    expect(screen.getByLabelText(/toogle dropdown/)).toBeInTheDocument()
  })

  it('should handle open/close dropdown', () => {
    const dropDownContent = screen.getByText(/content/i).parentElement!

    expect(dropDownContent).toHaveStyle({ opacity: 0 })
    expect(dropDownContent.getAttribute('aria-hidden')).toBe('true')

    userEvent.click(screen.getByLabelText(/toogle dropdown/))

    expect(dropDownContent).toHaveStyle({ opacity: 1 })
    expect(dropDownContent.getAttribute('aria-hidden')).toBe('false')
  })

  it('should handle open/close dropdown when overlay clicking on overlay', () => {
    const content = screen.getByText(/content/i).parentElement!
    const overlay = content.nextElementSibling!

    userEvent.click(screen.getByLabelText(/toogle dropdown/))

    expect(overlay).toHaveStyle({ opacity: 1 })
    expect(overlay.getAttribute('aria-hidden')).toBe('false')

    userEvent.click(overlay)

    expect(overlay).toHaveStyle({ opacity: 0 })
    expect(overlay.getAttribute('aria-hidden')).toBe('true')
  })
})
