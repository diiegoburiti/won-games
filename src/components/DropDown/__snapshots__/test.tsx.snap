// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDown /> should handle open/close dropdown 1`] = `
.c1 {
  cursor: pointer;
  color: #FAFAFA;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2.4rem;
  z-index: 50;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FAFAFA;
  color: #030517;
  margin-top: 2.4rem;
  position: absolute;
  right: 0;
  z-index: 50;
}

.c3::before {
  content: '';
  position: absolute;
  border-right: 1.2rem solid transparent;
  border-left: 1.2rem solid transparent;
  border-bottom: 1.2rem solid #FAFAFA;
  top: -1.2rem;
  right: 2.4rem;
}

.c5 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 30;
  background: rgba(0,0,0,0.5);
}

.c0 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c0 .c2,
.c0 .c4 {
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-2rem);
  -ms-transform: translateY(-2rem);
  transform: translateY(-2rem);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h5
      aria-label="toogle dropdown"
    >
      Click here
    </h5>
  </div>
  <div
    aria-hidden="true"
    class="c2 c3"
  >
    <span>
      content
    </span>
  </div>
  <div
    aria-hidden="true"
    class="c4 c5"
  />
</div>
`

exports[
  `<DropDown /> should handle open/close dropdown when overlay clicking on overlay 1`
] = `
.c1 {
  cursor: pointer;
  color: #FAFAFA;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2.4rem;
  z-index: 50;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FAFAFA;
  color: #030517;
  margin-top: 2.4rem;
  position: absolute;
  right: 0;
  z-index: 50;
}

.c3::before {
  content: '';
  position: absolute;
  border-right: 1.2rem solid transparent;
  border-left: 1.2rem solid transparent;
  border-bottom: 1.2rem solid #FAFAFA;
  top: -1.2rem;
  right: 2.4rem;
}

.c5 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 30;
  background: rgba(0,0,0,0.5);
}

.c0 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c0 .c2,
.c0 .c4 {
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-2rem);
  -ms-transform: translateY(-2rem);
  transform: translateY(-2rem);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h5
      aria-label="toogle dropdown"
    >
      Click here
    </h5>
  </div>
  <div
    aria-hidden="true"
    class="c2 c3"
  >
    <span>
      content
    </span>
  </div>
  <div
    aria-hidden="true"
    class="c4 c5"
  />
</div>
`

exports[`<DropDown /> should render title 1`] = `
.c1 {
  cursor: pointer;
  color: #FAFAFA;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2.4rem;
  z-index: 50;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FAFAFA;
  color: #030517;
  margin-top: 2.4rem;
  position: absolute;
  right: 0;
  z-index: 50;
}

.c3::before {
  content: '';
  position: absolute;
  border-right: 1.2rem solid transparent;
  border-left: 1.2rem solid transparent;
  border-bottom: 1.2rem solid #FAFAFA;
  top: -1.2rem;
  right: 2.4rem;
}

.c5 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 30;
  background: rgba(0,0,0,0.5);
}

.c0 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c0 .c2,
.c0 .c4 {
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-2rem);
  -ms-transform: translateY(-2rem);
  transform: translateY(-2rem);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h5
      aria-label="toogle dropdown"
    >
      Click here
    </h5>
  </div>
  <div
    aria-hidden="true"
    class="c2 c3"
  >
    <span>
      content
    </span>
  </div>
  <div
    aria-hidden="true"
    class="c4 c5"
  />
</div>
`
